test/test_python_errors.py

```diff
@@ -375,3 +375,15 @@ def test_repeated_kwarg():
         _get_error_list("f(q=1, q=2)", version="3.9")[0].message
         == "SyntaxError: keyword argument repeated: q"
     )
+
+
+@pytest.mark.parametrize(
+    'source', [
+        'a(a for a in b,)'
```

Comment:
I think this case should actually pass:

```
>>> str(a for a in [1],)
'<generator object <genexpr> at 0x7fecc48320a0>'
```

You should probably add a second param `has_errors` and parametrize with that as well. Please also add a simple case like `a(a for a in b)`.

Parso is sometimes not tested extremely well. The reason for this is that for a very long time Parso was part of Jedi and didn't really have its own tests. Nowadays there are at least a few tests that help, but they are not extensive at all.

Reply:
interesting.
```
 $ python
Python 3.8.0+ (heads/3.8:b9e5547f58, Nov 28 2019, 19:18:03) 
[GCC 9.2.1 20191008] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> a(a for a in b,)
  File "<stdin>", line 1
SyntaxError: Generator expression must be parenthesized
>>> 
```

Reply:
```
 $ python3.10
Python 3.10.0a0 (heads/bpo-xxxxx:f2947e354c, May 21 2020, 18:54:57) 
[GCC 9.2.1 20191008] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> str(a for a in [1],)
  File "<stdin>", line 1
    str(a for a in [1],)
        ^
SyntaxError: Generator expression must be parenthesized
```

Reply:
Looks like a python2 issue, will add a condition

Reply:
BTW: Just in general, I don't think 3.5/2.7 support is important anymore. So don't put in too much effort. This is just something I found by accident, because 3.5 was open. IMO I wouldn't even fix it, because people that do that will soon be screwed anyway :)

I guess I was more worried that `str(a for a in [1])` would also not pass.

Reply:
> I guess I was more worried that str(a for a in [1]) would also not pass.
```
>>> get_errors("str(a for a in b)")
[]
```